File: backend/scripts/verify_sqlite.py
"""Verify SQLite tables and data format. Run: python scripts/verify_sqlite.py"""
import sys
import os
from datetime import datetime, timedelta

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

import sqlite3
import json
from database.database import init_db, db_connection

# Mock data matching Express mocks (for seeding when Express hasn't run)
GEO_EVENTS = [
    ("2026-02-06 14:23:00", "sanctions", "CRITICAL", "New EU Sanctions on Russian Energy Sector",
     "European Union announces comprehensive sanctions...", "Russia", 125, "EU Official Journal"),
    ("2026-02-06 14:18:00", "trade", "HIGH", "Belarus Trade Restrictions Extended",
     "US Treasury extends trade restrictions...", "Belarus", 45, "OFAC"),
]
REC_TASKS = [
    ("REC-001", "EU Sanctions Update", "Automated Policy Monitor", "completed",
     1247, 125, 125, "2026-02-06 14:23:00", "2026-02-06 14:23:45", 3200, "AI Engine", "critical"),
]

def seed_mocks(conn):
    if conn.execute("SELECT COUNT(*) FROM geo_events").fetchone()[0] == 0:
        for row in GEO_EVENTS:
            conn.execute(
                "INSERT INTO geo_events (timestamp, type, severity, title, description, country, affected_transactions, source) VALUES (?,?,?,?,?,?,?,?)",
                row,
            )
        print("Seeded geo_events")
    if conn.execute("SELECT COUNT(*) FROM reconciliation_tasks").fetchone()[0] == 0:
        for row in REC_TASKS:
            conn.execute(
                """INSERT OR IGNORE INTO reconciliation_tasks (id, event_type, triggered_by, status, transactions_scanned, transactions_flagged, transactions_reconciled, start_time, completion_time, estimated_savings, assigned_to, priority) VALUES (?,?,?,?,?,?,?,?,?,?,?,?)""",
                row,
            )
        print("Seeded reconciliation_tasks")

def main():
    init_db()
    with db_connection() as conn:
        seed_mocks(conn)

    from config.config import settings
    db_path = settings.DATABASE_URL.replace("sqlite:///", "").strip("/") or "politifolio.db"
    if not os.path.exists(db_path):
        print("DB file not found at", db_path)
        return

    conn = sqlite3.connect(db_path)
    conn.row_factory = sqlite3.Row

    for t in ["geo_events", "reconciliation_tasks", "key_events"]:
        c = conn.execute(f"SELECT COUNT(*) FROM {t}").fetchone()[0]
        print(f"{t}: {c} rows")

    # geo_events schema & sample
    cols = [r[1] for r in conn.execute("PRAGMA table_info(geo_events)").fetchall()]
    print("\ngeo_events columns:", cols)
    expected_cols = ["id", "timestamp", "type", "severity", "title", "description", "country", "affected_transactions", "source"]
    print("  Expected:", expected_cols)
    print("  Match:", set(expected_cols) <= set(cols))

    row = conn.execute("SELECT * FROM geo_events LIMIT 1").fetchone()
    if row:
        d = dict(row)
        print("  Sample:", json.dumps(d, default=str, indent=2))

    # reconciliation_tasks schema & sample
    cols = [r[1] for r in conn.execute("PRAGMA table_info(reconciliation_tasks)").fetchall()]
    print("\nreconciliation_tasks columns:", cols)

    row = conn.execute("SELECT * FROM reconciliation_tasks LIMIT 1").fetchone()
    if row:
        d = dict(row)
        print("  Sample:", json.dumps(d, default=str, indent=2))

    # key_events schema
    cols = [r[1] for r in conn.execute("PRAGMA table_info(key_events)").fetchall()]
    print("\nkey_events columns:", cols)

    # Format check: frontend expects camelCase (affectedTransactions, eventType, etc.)
    print("\nFormat check for frontend:")
    print("  geo_events DB uses snake_case (affected_transactions)")
    print("  Express sqlite.service maps to camelCase (affectedTransactions) in getGeoEvents()")
    print("  reconciliation_tasks DB uses snake_case (event_type, triggered_by)")
    print("  Express sqlite.service maps to camelCase (eventType, triggeredBy) in getReconciliationTasks()")

    conn.close()
    print("\nVerification complete.")

if __name__ == "__main__":
    main()

File: backend/scripts/verify_sqlite.ts
/**
 * Verify SQLite tables are populated and data format is correct.
 * Run: npx tsx scripts/verify_sqlite.ts
 */
import Database from 'better-sqlite3'
import { dirname, join } from 'path'
import { fileURLToPath } from 'url'
import { getEventsMock } from '../src/mocks/events.mock.js'
import { getReconciliationTasksMock } from '../src/mocks/reconciliation-tasks.mock.js'
import { initTables, seedFromMocks, getGeoEvents, getReconciliationTasks } from '../src/services/sqlite.service.js'

const __dirname = dirname(fileURLToPath(import.meta.url))
const DB_PATH = join(__dirname, '..', 'politifolio.db')

function main() {
  console.log('DB path:', DB_PATH)
  console.log('')

  // 1. Init and seed
  initTables()
  seedFromMocks(getEventsMock(), getReconciliationTasksMock())

  // 2. Query raw DB
  const db = new Database(DB_PATH, { readonly: true })
  const geoCount = (db.prepare('SELECT COUNT(*) as c FROM geo_events').get() as { c: number }).c
  const recCount = (db.prepare('SELECT COUNT(*) as c FROM reconciliation_tasks').get() as { c: number }).c
  const keyCount = (db.prepare('SELECT COUNT(*) as c FROM key_events').get() as { c: number }).c
  console.log('Row counts:')
  console.log('  geo_events:', geoCount)
  console.log('  reconciliation_tasks:', recCount)
  console.log('  key_events:', keyCount)
  console.log('')

  // 3. Sample geo_events
  if (geoCount > 0) {
    const row = db.prepare('SELECT * FROM geo_events LIMIT 1').get() as Record<string, unknown>
    console.log('geo_events sample (raw):', JSON.stringify(row, null, 2))
    const expected = ['id', 'timestamp', 'type', 'severity', 'title', 'description', 'country', 'affected_transactions', 'source']
    const hasAll = expected.every((k) => k in row)
    console.log('  Has all columns:', hasAll)
  }
  console.log('')

  // 4. Sample reconciliation_tasks
  if (recCount > 0) {
    const row = db.prepare('SELECT * FROM reconciliation_tasks LIMIT 1').get() as Record<string, unknown>
    console.log('reconciliation_tasks sample (raw):', JSON.stringify(row, null, 2))
  }
  console.log('')

  // 5. Service layer format (camelCase for frontend)
  const events = getGeoEvents()
  const tasks = getReconciliationTasks()
  console.log('Service getGeoEvents() format:', events?.[0] ? JSON.stringify(events[0], null, 2) : 'null')
  console.log('  affectedTransactions (camelCase):', events?.[0] && 'affectedTransactions' in events[0])
  console.log('')
  console.log('Service getReconciliationTasks() format:', tasks?.[0] ? JSON.stringify(tasks[0], null, 2) : 'null')
  console.log('  eventType, triggeredBy (camelCase):', tasks?.[0] && 'eventType' in tasks[0] && 'triggeredBy' in tasks[0])

  db.close()
  console.log('')
  console.log('Verification complete.')
}

main()
